### PR TITLE
CVSL-2153 add hdc curfew times migration, entity and model

### PR DESCRIPTION
**Pull request by louisenorris**

### CODE REVIEW

**Review by andrewrlee — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/HdcService.kt [hunk 1]
```diff
@@ -1,23 +1,44 @@
...
+      .orElseThrow { EntityNotFoundException("No licence data found for $licenceId") }
 
-    return HdcLicenceData(
-      curfewAddress = licenceData.curfewAddress,
-      firstNightCurfewHours = licenceData.firstNightCurfewHours,
-      curfewHours = licenceData.curfewHours,
-    )
+    val curfewTimes = hdcCurfewTimesRepository.findByLicenceId(licenceId)
+
+    val licenceData = this.hdcApiClient.getByBookingId(licence.bookingId!!)
```
Comment (louisenorris) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/HdcService.kt:26]: I think we still need to call the `hdcApiClient` every time as we don't yet store `curfewAddress` or `firstNightCurfewHours` in the CVL db

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/HdcService.kt [hunk 2]
```diff
@@ -1,23 +1,44 @@
...
+    val licence = licenceRepository
+      .findById(licenceId)
+      .orElseThrow { EntityNotFoundException("No licence data found for $licenceId") }
 
-    return HdcLicenceData(
-      curfewAddress = licenceData.curfewAddress,
-      firstNightCurfewHours = licenceData.firstNightCurfewHours,
-      curfewHours = licenceData.curfewHours,
-    )
+    val curfewTimes = hdcCurfewTimesRepository.findByLicenceId(licenceId)
```
Comment (louisenorris) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/HdcService.kt:24]: Maybe it is now unnecessary to call the `hdcCurfewTimesRepository` for `curfewTimes` as they will be on the returned `licence` object if present in the CVL db?

Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/HdcService.kt:24]: yes, better just to load it from licence itself

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/repository/HdcCurfewTimesRepository.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository
+
+import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.stereotype.Repository
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.entity.HdcCurfewTimes
+
+@Repository
+interface HdcCurfewTimesRepository : JpaRepository<HdcCurfewTimes, Long> {
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/repository/HdcCurfewTimesRepository.kt:8]: so might be able to remove this if its not used

Comment (louisenorris) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/repository/HdcCurfewTimesRepository.kt]: removed - thank you!

**Review by justin-gardiner — APPROVED** [review]
LGTM

**Pull request merged**